# Import libraries

In [1]:
#Importing libraries
import pandas as pd
import numpy as np
import os

# Import dataset

In [2]:
#define the path of the main folder
path = r'/Users/gil/Downloads/World Happiness'

In [3]:
#import df_combined.csv
df_combined = pd.read_csv(os.path.join(path, 'Data', 'Prepared data', 'Data combined World Happiness.csv'), index_col = False)

# Clean dataset

In [4]:
df_combined.head()

Country or region  Year  Happiness Rank  Happiness Score  \
0       Switzerland  2015               1            7.587   
1           Iceland  2015               2            7.561   
2           Denmark  2015               3            7.527   
3            Norway  2015               4            7.522   
4            Canada  2015               5            7.427   

   Economy (GDP per Capita)   Family  Health (Life Expectancy)  \
0                   1.39651  1.34951                   0.94143   
1                   1.30232  1.40223                   0.94784   
2                   1.32548  1.36058                   0.87464   
3                   1.45900  1.33095                   0.88521   
4                   1.32629  1.32261                   0.90563   

   Freedom to make life choices  Generosity  Trust (Government Corruption)  
0                       0.66557     0.29678                        0.41978  
1                       0.62877     0.43630                        0.14145  
2                       0.64938     0.34139                        0.48357  
3                       0.66973     0.34699                        0.36503  
4                       0.63297     0.45811                        0.32957

In [5]:
df_combined=df_combined.round(1)

In [6]:
df_combined.shape

(782, 10)

In [8]:
df_combined.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 782 entries, 0 to 781
Data columns (total 10 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Country or region              782 non-null    object 
 1   Year                           782 non-null    int64  
 2   Happiness Rank                 782 non-null    int64  
 3   Happiness Score                782 non-null    float64
 4   Economy (GDP per Capita)       782 non-null    float64
 5   Family                         782 non-null    float64
 6   Health (Life Expectancy)       782 non-null    float64
 7   Freedom to make life choices   782 non-null    float64
 8   Generosity                     782 non-null    float64
 9   Trust (Government Corruption)  781 non-null    float64
dtypes: float64(7), int64(2), object(1)
memory usage: 61.2+ KB


In [9]:
df_combined.columns

Index(['Country or region', 'Year', 'Happiness Rank', 'Happiness Score',
       'Economy (GDP per Capita)', 'Family', 'Health (Life Expectancy)',
       'Freedom to make life choices', 'Generosity',
       'Trust (Government Corruption)'],
      dtype='object')

In [17]:
#check for missing values
df_combined.isnull().sum()

Country or region                0
Year                             0
Happiness Rank                   0
Happiness Score                  0
Economy (GDP per Capita)         0
Family                           0
Health (Life Expectancy)         0
Freedom to make life choices     0
Generosity                       0
Trust (Government Corruption)    1
Financial status                 0
Government status                0
dtype: int64

In [18]:
df_dupes_combined = df_combined[df_combined.duplicated()]

In [19]:
df_dupes_combined

Empty DataFrame
Columns: [Country or region, Year, Happiness Rank, Happiness Score, Economy (GDP per Capita), Family, Health (Life Expectancy), Freedom to make life choices, Generosity, Trust (Government Corruption), Financial status, Government status]
Index: []

# Statistical analysis

In [20]:
df_combined.describe()

Year  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
count   782.000000      782.000000       782.000000                782.000000   
mean   2016.993606       78.698210         5.381202                  0.917263   
std       1.417364       45.182384         1.126786                  0.407354   
min    2015.000000        1.000000         2.700000                  0.000000   
25%    2016.000000       40.000000         4.500000                  0.600000   
50%    2017.000000       79.000000         5.300000                  1.000000   
75%    2018.000000      118.000000         6.200000                  1.200000   
max    2019.000000      158.000000         7.800000                  2.100000   

           Family  Health (Life Expectancy)  Freedom to make life choices  \
count  782.000000                782.000000                    782.000000   
mean     1.078261                  0.610102                      0.411509   
std      0.331161                  0.248813                      0.155877   
min      0.000000                  0.000000                      0.000000   
25%      0.900000                  0.400000                      0.300000   
50%      1.100000                  0.600000                      0.400000   
75%      1.300000                  0.800000                      0.500000   
max      1.600000                  1.100000                      0.700000   

       Generosity  Trust (Government Corruption)  
count  782.000000                     781.000000  
mean     0.219054                       0.127017  
std      0.126946                       0.112432  
min      0.000000                       0.000000  
25%      0.100000                       0.100000  
50%      0.200000                       0.100000  
75%      0.300000                       0.200000  
max      0.800000                       0.600000

# Create a financial status column

Highest GDP is 2.10, lowest is 0. I divided it into 3 categories (0-0.7, 0.7-1.4, and 1.4-2.1), and assigned it different walues. 

In [10]:
def Financial_label (row):

  if row['Economy (GDP per Capita)'] <= 0.7:
    return 'Financially struggling'
  elif (row['Economy (GDP per Capita)'] > 0.7) and (row['Economy (GDP per Capita)'] < 1.4):
    return 'Financially average'
  elif row['Economy (GDP per Capita)'] >= 1.4:
    return 'Financially well-off'
  else: return 'Not enough data'

In [11]:
#apply the function 
df_combined['Financial status'] = df_combined.apply(Financial_label, axis=1)

In [12]:
df_combined.head(30)

Country or region  Year  Happiness Rank  Happiness Score  \
0            Switzerland  2015               1              7.6   
1                Iceland  2015               2              7.6   
2                Denmark  2015               3              7.5   
3                 Norway  2015               4              7.5   
4                 Canada  2015               5              7.4   
5                Finland  2015               6              7.4   
6            Netherlands  2015               7              7.4   
7                 Sweden  2015               8              7.4   
8            New Zealand  2015               9              7.3   
9              Australia  2015              10              7.3   
10                Israel  2015              11              7.3   
11            Costa Rica  2015              12              7.2   
12               Austria  2015              13              7.2   
13                Mexico  2015              14              7.2   
14         United States  2015              15              7.1   
15                Brazil  2015              16              7.0   
16            Luxembourg  2015              17              6.9   
17               Ireland  2015              18              6.9   
18               Belgium  2015              19              6.9   
19  United Arab Emirates  2015              20              6.9   
20        United Kingdom  2015              21              6.9   
21                  Oman  2015              22              6.9   
22             Venezuela  2015              23              6.8   
23             Singapore  2015              24              6.8   
24                Panama  2015              25              6.8   
25               Germany  2015              26              6.8   
26                 Chile  2015              27              6.7   
27                 Qatar  2015              28              6.6   
28                France  2015              29              6.6   
29             Argentina  2015              30              6.6   

    Economy (GDP per Capita)  Family  Health (Life Expectancy)  \
0                        1.4     1.3                       0.9   
1                        1.3     1.4                       0.9   
2                        1.3     1.4                       0.9   
3                        1.5     1.3                       0.9   
4                        1.3     1.3                       0.9   
5                        1.3     1.3                       0.9   
6                        1.3     1.3                       0.9   
7                        1.3     1.3                       0.9   
8                        1.3     1.3                       0.9   
9                        1.3     1.3                       0.9   
10                       1.2     1.2                       0.9   
11                       1.0     1.2                       0.9   
12                       1.3     1.3                       0.9   
13                       1.0     0.9                       0.8   
14                       1.4     1.2                       0.9   
15                       1.0     1.2                       0.7   
16                       1.6     1.2                       0.9   
17                       1.3     1.4                       0.9   
18                       1.3     1.3                       0.9   
19                       1.4     1.1                       0.8   
20                       1.3     1.3                       0.9   
21                       1.4     1.1                       0.8   
22                       1.0     1.3                       0.7   
23                       1.5     1.0                       1.0   
24                       1.1     1.2                       0.8   
25                       1.3     1.3                       0.9   
26                       1.1     1.1                       0.9   
27                       1.7     1.1                       0.8   
28             

# Create Government status column

In [13]:
def Government_label (row):

  if row['Trust (Government Corruption)'] <= 0.2:
    return 'Low government trust'
  elif (row['Trust (Government Corruption)'] > 0.2) and (row['Trust (Government Corruption)'] < 0.4):
    return 'Average government trust'
  elif row['Trust (Government Corruption)'] >= 0.4:
    return 'High government trust'
  else: return 'Not enough data'

In [14]:
#apply the function 
df_combined['Government status'] = df_combined.apply(Government_label, axis=1)

In [15]:
df_combined.head()

Country or region  Year  Happiness Rank  Happiness Score  \
0       Switzerland  2015               1              7.6   
1           Iceland  2015               2              7.6   
2           Denmark  2015               3              7.5   
3            Norway  2015               4              7.5   
4            Canada  2015               5              7.4   

   Economy (GDP per Capita)  Family  Health (Life Expectancy)  \
0                       1.4     1.3                       0.9   
1                       1.3     1.4                       0.9   
2                       1.3     1.4                       0.9   
3                       1.5     1.3                       0.9   
4                       1.3     1.3                       0.9   

   Freedom to make life choices  Generosity  Trust (Government Corruption)  \
0                           0.7         0.3                            0.4   
1                           0.6         0.4                            0.1   
2                           0.6         0.3                            0.5   
3                           0.7         0.3                            0.4   
4                           0.6         0.5                            0.3   

       Financial status         Government status  
0  Financially well-off     High government trust  
1   Financially average      Low government trust  
2   Financially average     High government trust  
3  Financially well-off     High government trust  
4   Financially average  Average government trust

# Export dataset

In [16]:
#export df_combined
df_combined.to_csv(os.path.join(path, 'Data', 'Prepared data', 'combined_data_checked.csv'))